scripts/wasmtestreport.py

```diff
@@ -725,6 +730,16 @@ def main():
         "non_deterministic": get_empty_result()
     }
 
+    # Prepare artifacts root
+    created_temp_dir = False
+    if artifacts_dir_arg:
+        artifacts_root = artifacts_dir_arg.resolve()
```

Comment:
if the artifacts dir is requested but not writable, this will behave unexpectedly.  Perhaps a try/except would help on it to make sure?